### PR TITLE
Adjust allowed version of url to avoid issues vendoring into m-c

### PR DESCRIPTION
**Pull request by mhammond**

### CODE REVIEW

**Comment by mhammond** [comment]
FTR, on top of bug 1770894 there are no new dupes

**Comment by mhammond** [comment]
This appears to be all we need on top of bug 1770894, so requesting review now, but I'll probably hold off merging until that lands and I'm in a position to put up the m-c patch.

File: components/support/sync15-traits/Cargo.toml
Comment (glandium) [inline components/support/sync15-traits/Cargo.toml:17]: Technically, this allows 2.2. And it's fine.

Comment (mhammond) [reply components/support/sync15-traits/Cargo.toml:17]: I'm not sure what you mean here - maybe the comment is misleading?

The problem IIRC is that from 2.1 -> 2.2 they added `Into<String>` (or something like that, I last struck it mid last year) but also dropped the "old" way of getting a string, which I think was a mistake on their part as it broke semver. I can dig up the details if you are interested.

Comment (mhammond) [reply components/support/sync15-traits/Cargo.toml:17]: doh, sorry, I do see what you mean, yeah

Comment (mhammond) [reply components/support/sync15-traits/Cargo.toml]: So I don't think there's anything I can really do to stop these crates from using features in 2.2, so I guess we'll just discover if they do as we vendor new versions.

Comment (glandium) [reply components/support/sync15-traits/Cargo.toml]: You can do >=2.1,<2.2

Comment (mhammond) [reply components/support/sync15-traits/Cargo.toml]: But then we will end up with a dupe of that crate in our mobile builds (some crates that aren't in m-c *do* need 2.2)

**Review by tarikeshaq — APPROVED** [review]

**Pull request merged**
